tests/test_compare.py

```diff
@@ -324,3 +324,15 @@ def test_trackmodelresult_and_trackobservation_uses_model_name():
 def test_item_selection_items_are_unique():
     with pytest.raises(ValueError):
         ItemSelection(obs="foo", model=["foo", "bar"], aux=["baz"])
+
+
+def test_save_comparercollection(o1, o3, tmp_path):
```

Comment:
Maybe, I suppose the entire test suite needs to be (re-)organized. I think it would make sense to split tests into unit tests and integration tests and end-to-end tests, or something along those lines.